airflow/api_connexion/webserver_auth.py

```diff
@@ -0,0 +1,95 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import datetime, timedelta
+from functools import wraps
+from typing import Callable, TypeVar, cast
+
+import jwt
+from flask import Response, current_app, request
+from flask_appbuilder.const import AUTH_LDAP
+from flask_login import login_user
+
+from airflow.configuration import conf
+
+SECRET = conf.get("webserver", "secret_key")
+T = TypeVar("T", bound=Callable)  # pylint: disable=invalid-name
+
+
+def encode_auth_token(user_id):
+    """Generate authentication token"""
+    expire_time = conf.getint("webserver", "session_lifetime_minutes")
```

Comment:
The default is 30 days. It's a bit too long for an access token. The access token should be generated for a short period of time, e.g. 1 hour, and then refreshed regulary with a refresh token. My main concerns are what happens if the token is stolen and the user changes the password? Will this token still work?

Reply:
Yeah, having the password change def should invalidate the token -- that could only work for "local" (in DB) users I think, but it's still worth doing.